drivers/serial/uart_nrfx_uarte.c

```diff
@@ -2888,15 +2896,23 @@ static void uarte_nrfx_irq_rx_enable(const struct device *dev)
 {
 	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
 
-	nrf_uarte_int_enable(uarte, NRF_UARTE_INT_ENDRX_MASK);
+	if (!nrf_uarte_int_enable_check(uarte, NRF_UARTE_INT_ENDRX_MASK)) {
+		pm_device_runtime_get(dev);
+
+		nrf_uarte_int_enable(uarte, NRF_UARTE_INT_ENDRX_MASK);
+	}
 }
 
 /** Interrupt driven receiver disabling function */
 static void uarte_nrfx_irq_rx_disable(const struct device *dev)
 {
 	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
 
-	nrf_uarte_int_disable(uarte, NRF_UARTE_INT_ENDRX_MASK);
+	if (nrf_uarte_int_enable_check(uarte, NRF_UARTE_INT_ENDRX_MASK)) {
+		pm_device_runtime_put(dev);
```

Comment:
i think that it should be `pm_device_runtime_put_async(dev, K_NO_WAIT);` as it may happen from the interrupt context. For ISR_SAFE case that will be the same thing.